ui/apps/platform/cypress/integration/vulnerabilities/nodeCves/NodeCve.helpers.ts

```diff
@@ -1,27 +1,87 @@
+import type { RouteHandler, RouteMatcherOptions } from 'cypress/types/net-stubbing';
+
 import { graphql } from '../../../constants/apiEndpoints';
 import {
     getRouteMatcherMapForGraphQL,
     interactAndWaitForResponses,
 } from '../../../helpers/request';
+import { visit, visitWithStaticResponseForPermissions } from '../../../helpers/visit';
 
-export function mockOverviewNodeCveListRequest() {
-    const opname = 'getNodeCVEs';
-    cy.intercept(
-        { method: 'POST', url: graphql(opname) },
-        { fixture: `vulnerabilities/nodeCves/${opname}.json` }
-    ).as(opname);
-}
+export const nodeCveBaseUrl = '/main/vulnerabilities/node-cves/cves';
+
+// Source of truth for keys in routeMatcherMap and staticResponseMap objects.
+// Overview page
+export const getNodesOpname = 'getNodes';
+export const getNodeCvesOpname = 'getNodeCVEs';
+
+// Node CVE page
+export const getNodeCveMetadataOpname = 'getNodeCVEMetadata';
+
+// Node page
+export const getNodeMetadataOpname = 'getNodeMetadata';
+export const getNodeVulnSummaryOpname = 'getNodeVulnSummary';
+export const getNodeVulnerabilitiesOpname = 'getNodeVulnerabilities';
 
-export function mockOverviewNodeListRequest() {
-    const opname = 'getNodes';
-    cy.intercept(
-        { method: 'POST', url: graphql(opname) },
-        { fixture: `vulnerabilities/nodeCves/${opname}.json` }
-    ).as(opname);
+export const routeMatcherMapForNodes = {
+    [getNodesOpname]: {
+        method: 'POST',
+        url: graphql(getNodesOpname),
+    },
+};
+
+export const routeMatcherMapForNodeCves = {
+    [getNodeCvesOpname]: {
+        method: 'POST',
+        url: graphql(getNodeCvesOpname),
+    },
+};
+
+export const routeMatcherMapForNodeCveMetadata = {
+    [getNodeCveMetadataOpname]: {
+        method: 'POST',
+        url: graphql(getNodeCveMetadataOpname),
+    },
+};
+
+export const routeMatcherMapForNodePage = {
+    [getNodeMetadataOpname]: {
+        method: 'POST',
+        url: graphql(getNodeMetadataOpname),
+    },
+    [getNodeVulnSummaryOpname]: {
+        method: 'POST',
+        url: graphql(getNodeVulnSummaryOpname),
+    },
+    [getNodeVulnerabilitiesOpname]: {
+        method: 'POST',
+        url: graphql(getNodeVulnerabilitiesOpname),
+    },
+};
+
+// visit
+export function visitNodeCveOverviewPage(
+    routeMatcherMap?: Record<string, RouteMatcherOptions>,
+    staticResponseMap?: Record<string, RouteHandler>
```

Comment:
Existing issue in many function that I have written, but for TypeScript versus JavaScript file, is there a straightforward pattern for us to write function declaration with explicit requirement that `routeMatcherMap` is required if function call has `staticResponseMap` argument?

Reply:
Straightforward... maybe, maybe-not, but there is a valid pattern that can accomplish this: https://github.com/stackrox/stackrox/pull/12133#discussion_r1695508681

We would need to refactor some of the surrounding code to enforce this pattern though. 

I also haven't found a good way to make this function type declaration more reusable, unless we switch to an options object style API for these helpers:
```ts
type ResponseMapArgs<RouteKey extends string> = {
    routeMatcherMap: Record<RouteKey, RouteMatcherOptions>;
    staticResponseMap: Partial<Record<RouteKey, RouteMatcherOptions>>;
};

export function interceptAndWatchRequests<RouteKey extends string>({
    routeMatcherMap,
    staticResponseMap,
}: ResponseMapArgs<RouteKey>) {}
```